python/foglamp/plugins/storage/postgres/backup_restore/backup_postgres.py

```diff
@@ -47,37 +56,273 @@
+                                self,
+                                limit: int,
+                                skip: int,
+                                status: [lib.BackupStatus, None],
+                                sort_order: lib.SortOrder = lib.SortOrder.DESC) -> list:
+
+        """ Returns a list of backups is returned sorted in chronological order with the most recent backup first.
+
+        Args:
+            limit: int - limit the number of backups returned to the number given
+            skip: int - skip the number of backups specified before returning backups-
+                  this, in conjunction with the limit option, allows for a paged interface to be built
+            status: lib.BackupStatus - limit the returned backups to those only with the specified status,
+                    None = retrieves information for all the backups
+            sort_order: lib.SortOrder - Defines the order used to present information, DESC by default
+
+        Returns:
+            backups_information: all the information available related to the requested backups
+
+        Raises:
+        """
+
+        if status is None:
+            payload = payload_builder.PayloadBuilder() \
+                .LIMIT(limit) \
+                .SKIP(skip) \
+                .ORDER_BY(['ts', sort_order]) \
+                .payload()
+        else:
+            payload = payload_builder.PayloadBuilder() \
+                .WHERE(['status', '=', status]) \
+                .LIMIT(limit) \
+                .SKIP(skip) \
+                .ORDER_BY(['ts', sort_order]) \
+                .payload()
+
+        backups_from_storage = self._storage.query_tbl_with_payload(lib.STORAGE_TABLE_BACKUPS, payload)
+
+        backups_information = backups_from_storage['rows']
+
+        return backups_information
+
+    def get_backup_details(self, backup_id: int) -> dict:
+        """ Returns the details of a backup
+
+        Args:
+            backup_id: int - the id of the backup to return
+
+        Returns:
+            backup_info: dict - Backup information :
+                id: An identifier for the backup
+                date: The date the backup started
+                status: The status of the backup; one of running, failed, completed
+        Raises:
+            exceptions.DoesNotExist
+            exceptions.NotUniqueBackup
+
+        """
+
+        backup_information_complete = self.get_backup_details_complete(backup_id)
+
+        status_decoded = lib.BackupStatus.text[int(backup_information_complete['status'])]
+
+        backups_information = {
+                                'id': backup_information_complete['id'],
+                                'date': backup_information_complete['ts'],
+                                'status': status_decoded
+                                }
+
+        return backups_information
+
+    def get_backup_details_complete(self, backup_id: int) -> dict:
+        """ Returns the details of a backup
```

Comment:
Again I dont think we need to replicate the same function again get_backup_details for completed backups.

Reply:
above